CRM/Core/JobManager.php

```diff
@@ -67,13 +63,39 @@ public function execute($auth = TRUE) {
 
     // it's not asynchronous at this stage
     CRM_Utils_Hook::cron($this);
-    foreach ($this->jobs as $job) {
-      if ($job->is_active) {
-        if ($job->needsRunning()) {
-          $this->executeJob($job);
-        }
+
+    // Get a list of the jobs that have completed previously
+    $successfulJobs = Job::get(FALSE)
+      ->addWhere('is_active', '=', TRUE)
+      ->addWhere('last_run_end', 'IS NOT NULL')
```

Comment:
```suggestion
      ->addClause('OR', ['last_run', 'IS NULL'], ['last_run', '<=', 'last_run_end', TRUE])
```
Otherwise it ends up including the unsuccessful ones that have run at least once successfully. And then the check for last_run=NULL is for ones that have never even started.